suite-native/module-send-receive/src/screens/ReceiveModalScreen.tsx

```diff
@@ -8,48 +12,39 @@ import {
     StackProps,
 } from '@suite-native/navigation';
 
-import { sendReceiveContentType, SendReceiveContentType } from '../contentType';
-import { AddressGenerationStep } from '../components/AddressGenerationStep';
-import { AddressConfirmationStep } from '../components/AddressConfirmationStep';
-import { FreshAddressStep } from '../components/FreshAddressStep';
-
-const DEFAULT_CONTENT_TYPE = sendReceiveContentType.createNewAddressToReceive;
+import { ReceiveAddress } from '../components/ReceiveAddress';
+import { ReceiveTextHint } from '../components/ReceiveTextHint';
 
 export const ReceiveModalScreen = ({
     route,
     navigation,
 }: StackProps<RootStackParamList, RootStackRoutes.ReceiveModal>) => {
-    const [selectedAccountKey, setSelectedAccountKey] = useState<string>(route.params.accountKey);
-    const [contentType, setContentType] = useState<SendReceiveContentType>(DEFAULT_CONTENT_TYPE);
-
-    const handleClose = useCallback(() => {
-        navigation.goBack();
-        setSelectedAccountKey('');
-        setContentType(DEFAULT_CONTENT_TYPE);
-    }, [navigation]);
-
-    const sendReceiveContent: Record<SendReceiveContentType, ReactNode> = useMemo(
-        () => ({
-            [sendReceiveContentType.createNewAddressToReceive]: (
-                <AddressGenerationStep
-                    accountKey={selectedAccountKey}
-                    onChangeContentType={setContentType}
-                />
-            ),
-            [sendReceiveContentType.confirmNewAddressToReceive]: (
-                <AddressConfirmationStep
-                    accountKey={selectedAccountKey}
-                    onChangeContentType={setContentType}
-                />
-            ),
-            [sendReceiveContentType.generatedAddressToReceive]: (
-                <FreshAddressStep accountKey={selectedAccountKey} onClose={handleClose} />
-            ),
-        }),
-        [handleClose, selectedAccountKey],
+    const [addressIsVisible, setAddressIsVisible] = useState(false);
+    const { accountKey } = route.params;
+    const account = useSelector((state: AccountsRootState) =>
+        selectAccountByKey(state, accountKey),
     );
 
+    if (!account) return null;
```

Comment:
Always try to use at least simple error message, it will help us debug even if this should never happen. I think we have some ErrorMessage component for this stuff.

Reply:
fixed c3142a603